pkg/sdk/pipes.go

```diff
@@ -0,0 +1,166 @@
+package sdk
+
+import (
+	"context"
+	"time"
+)
+
+type Pipes interface {
+	// Create creates a pipe.
+	Create(ctx context.Context, opts *PipeCreateOptions) error
+	// Alter modifies an existing pipe.
+	Alter(ctx context.Context, opts *PipeAlterOptions) error
+	// Drop removes a pipe.
+	Drop(ctx context.Context, opts *PipeDropOptions) error
+	// Show returns a list of pipes.
+	Show(ctx context.Context, opts *PipeShowOptions) ([]*Pipe, error)
+	// ShowByID returns a pipe by ID.
+	ShowByID(ctx context.Context, id SchemaObjectIdentifier) (*Pipe, error)
+	// Describe returns the details of a pipe.
+	Describe(ctx context.Context, id SchemaObjectIdentifier) (*Pipe, error)
+}
+
+// PipeCreateOptions contains options for creating a new pipe in the system for defining the COPY INTO <table> statement
+// used by Snowpipe to load data from an ingestion queue into tables.
+//
+// Based on https://docs.snowflake.com/en/sql-reference/sql/create-pipe.
+type PipeCreateOptions struct {
+	create      bool                   `ddl:"static" sql:"CREATE"` //lint:ignore U1000 This is used in the ddl tag
+	OrReplace   *bool                  `ddl:"keyword" sql:"OR REPLACE"`
+	pipe        bool                   `ddl:"static" sql:"PIPE"` //lint:ignore U1000 This is used in the ddl tag
+	IfNotExists *bool                  `ddl:"keyword" sql:"IF NOT EXISTS"`
+	name        SchemaObjectIdentifier `ddl:"identifier"` //lint:ignore U1000 This is used in the ddl tag
```

Comment:
I did it on purpose (from PR description: `interface methods without duplicated params (which reduces boilerplate and is cleaner)`).
IMO setting this name as a param, in the matter of fact setting any param, already contained in passed Options is weird thing and raises complexity.

Reply:
Right, so my understanding is that parameters which are required are present in the parameters list separately, to make sure caller of the function provides them.
It is rather clunky though. One way of dealing with this problem is to not use `Opts` structs directly in the interface, as it's really an implementation detail. Its structure depends solely on the SQL it's going to be converted to, so maybe using a separate representation for the arguments inside the interface functions and then translating it to the `Opts` struct would be helpful. That would also allow us to be more strict with validating the input at compile time and would make for a nicer interface for the sdk's users.

Reply:
I'm fine with setting name in struct, but I meant that It cannot be set in any way because 1. It's not exported and 2. it's not set in the generic function, so we should either export it or set it in the Create/Alter/.. functions, but we should also keep in mind is that every change like this will affect all of the resources and they'll have to be changed to fit this style of setting ids

Reply:
ad 1 this is not yet connected to any caller, so it can be (will be) changed in following PR
ad 2 I think it's more of an open discussion: how do we want our interfaces look like? And then adjust to the same way.

To make it fly now, I can adjust to existing way, but I would like a discussion soon about the interfaces we'd like to have here (because there were multiple ideas like dtos and so).

Reply:
Done.